src/oop/object_interactions/medium/q2/MainTest.java

```diff
@@ -0,0 +1,165 @@
+            Object[][] arguments = {
+                    {name, String.class},
+                    {secret, String.class},
+                    {friends, String[].class}
+            };
+            Object classInstance = testClass.createInstance(arguments);
+            _assertEquals(name, testClass.getFieldValue(classInstance, myNameAttributeName),
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+            _assertEquals(secret, testClass.getFieldValue(classInstance, mySecretAttributeName),
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+            _assertArrayEquals(friends, testClass.getFieldValue(classInstance, friendNamesAttributeName),
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+            _assertArrayEquals(new String[friends.length], testClass.getFieldValue(classInstance, friendSecretsAttributeName),
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+        }
+
+        assertTrue(testClass.hasMethod(getNameMethodName, null, String.class, new String[]{"public"}),
+                String.format("You have modified the provided %s method method. Please revert it to the original state.", getNameMethodName));
+        assertTrue(testClass.hasMethod(getSecretMethodName, null, String.class, new String[]{"public"}),
+                String.format("You have modified the provided %s method method. Please revert it to the original state.", getSecretMethodName));
+
+    }
+
+    private static Stream<Arguments> hearSecretInputProvider() {
+        return Stream.of(
+                Arguments.of("Sam", "Sam always cheat on exams", new String[]{"Bob", "Ali"}, "Bob", "Bob has a candy"),
+                Arguments.of("John", "John is in love with Alice", new String[]{"Alice", "Bob"}, "Alice", "Alice is in love with John"),
+                Arguments.of("Kim", "Kim has broken her leg", new String[]{"Ali", "John"}, "Ali", "Ali broke Kim's leg")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("hearSecretInputProvider")
+    public void correctHearSecretMethod(String name, String secret, String[] allFriends, String friend, String friendSecret) throws Throwable {
+        Object[][] arguments = {
+                {name, String.class},
+                {secret, String.class},
+                {allFriends, String[].class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        Object[][] methodArguments = {
+                {friend, String.class},
+                {friendSecret, String.class}
+        };
+        Object output = testClass.callMethod(hearSecretMethodName, methodArguments, new String[]{"public"}, classInstance);
```

Comment:
this should be split up into multiple asserts to check the name, arguments, and visibility seperately